src/main/java/org/aksw/simba/squirrel/analyzer/impl/RdfAutomata.java

```diff
@@ -0,0 +1,70 @@
+package org.aksw.simba.squirrel.analyzer.impl;
+
+import org.apache.jena.riot.Lang;
+import org.apache.jena.riot.RDFLanguages;
+
+/**
+        * Default implementation of a finite state machine.
+        * This class is immutable and thread-safe.
+        */
+
+public final class RdfAutomata implements FiniteStateMachine {
+
+    /**
+     * Current state.
+     */
+    private State current;
+    private Lang mimeType = RDFLanguages.RDFXML;
+
+    /**
+     * Ctor.
+     * @param initial Initial state of this machine.
+     */
+    private RdfAutomata(final State initial) {
+        this.current = initial;
+    }
+
+    public FiniteStateMachine switchState(final String c) {
+        return new RdfAutomata(this.current.transit(c));
```

Comment:
Why are you creating new automata representations? woudln't it be sufficient to move do something like
```java
this.current = this.current.transit(c);
return this;
```
Otherwise you are wasting time and memory by creating a lot of instances.